metroae

```diff
@@ -677,42 +677,85 @@ function destroy {
 
 function pull {
     debug ${FUNCNAME[0]}
-    if [[ ! -z $1 ]]; then
-        MAX_CONTAINER_VERSION=$1
-    fi
-    debug "${FUNCNAME[0]}: MAX_CONTAINER_VERSION: $MAX_CONTAINER_VERSION"
+    if [[ ! -z $1 ]] && [[ $1 == "--repo" ]]; then
+        REPO=$2
+        debug "${FUNCNAME[0]}: REPO: $REPO"
+        
+        set -e
 
-    set -e
+        echo ""
+        write_to_screen_and_script_log ">>> Pulling the MetroAE container image from Docker hub"
+        echo ""
 
-    echo ""
-    write_to_screen_and_script_log ">>> Pulling the MetroAE container image from Docker hub"
-    echo ""
-    if [[ -f $SCRIPT_LOG_FILE ]]; then
-        if [ -w "$SCRIPT_LOG_FILE" ]; then
-            debug "${FUNCNAME[0]}: Script log is writable"
-            sudo docker pull $METRO_AE_IMAGE:$MAX_CONTAINER_VERSION | tee -a $SCRIPT_LOG_FILE
+        if [[ -f $SCRIPT_LOG_FILE ]]; then
+            if [ -w "$SCRIPT_LOG_FILE" ]; then
+                debug "${FUNCNAME[0]}: Script log is writable"
+                sudo docker pull $REPO | tee -a $SCRIPT_LOG_FILE
+                ((status=$?+$PIPESTATUS))
+            else
+                if [ -z $METROAE_WRITE_WARNING_PRINTED ]; then
+                    print_write_permission_warning
+                    METROAE_WRITE_WARNING_PRINTED=1
+                    status=$?
+                fi
+                debug "${FUNCNAME[0]}: Script log is not writable"
+                sudo docker pull $REPO
+                status=$?
+            fi
         else
-            if [ -z $METROAE_WRITE_WARNING_PRINTED ]; then
-                print_write_permission_warning
-                METROAE_WRITE_WARNING_PRINTED=1
+            debug "${FUNCNAME[0]}: Script log is not found"
+            sudo docker pull $REPO
+            status=$?
+        fi
+
+        if [[ $status -ne 0 ]]; then
+            write_to_screen_and_script_log "Attempt to pull the $REPO MetroAE container image failed. Quitting."
+            exit 1
+        else
+            write_to_screen_and_script_log "Successfully pulled the MetroAE container image from specified repository."
+        fi
+        set -e
+    else
+        if [[ ! -z $1 ]]; then
```

Comment:
I'm not sure what this MAX_CONTAINER_VERSION is used for but don't we need to do the same logic in the case where we need a new repo.

An easier way to do all of this would be that, instead of using if/then/else,  we do the same logic as before but we change the `docker pull $METROAE_IMAGE:$MAX_CONTAINER_VERSION` with  `docker pull ${REPO}$METROAE_IMAGE:$MAX_CONTAINER_VERSION`
If the `--repo` option was not passed, then we leave REPO empty so the behaviour will be the same as before.
It he `--repo` option was passed, then we set REPO and we add a trailing slash at the end if it is not there.